yamux/src/connection/cleanup.rs

```diff
@@ -0,0 +1,68 @@
+use crate::connection::StreamCommand;
+use crate::ConnectionError;
+use futures::channel::mpsc;
+use futures::{ready, StreamExt};
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A [`Future`] that cleans up resources in case of an error.
```

Comment:
Just a thought, given that we already use the term _garbage collection_, how about renaming this to `GarbageCollector`? I might be missing the full picture here.

Reply:
This isn't quite the same as garbage collection. We run GC on every poll whereas this only runs on a fatal error of the connection.